components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/utils/APIDescriptionGenUtil.java

```diff
@@ -36,35 +36,48 @@
     private static final String DESCRIPTION = "Allows [1] request(s) per minute.";
 
     public static String generateDescriptionFromPolicy(OMElement policy) throws
-                                                                         AppManagementException {
+            AppManagementException {
         //Here as the method is about extracting some info from the policy. And it's not concern on compliance to
         // specification. So it just extract the required element.
+        int requestPerMinute;
+        try {
+            requestPerMinute = generateMaxCountFromPolicy(policy);
+            if (requestPerMinute >= 1) {
+                String description = DESCRIPTION.replaceAll("\\[1\\]", Integer.toString(requestPerMinute));
+                return description;
+            }
+            return DESCRIPTION;
+        } catch (NullPointerException npe) {
```

Comment:
We should not catch the NullPointerExceptions, instead we need to check for the != null and avoid the NullPointerException.